lib/matplotlib/axes/_axes.py

```diff
@@ -1756,18 +1755,19 @@ def xcorr(self, x, y, normed=True, detrend=mlab.detrend_none,
                              'positive < %d' % Nx)
 
         lags = np.arange(-maxlags, maxlags + 1)
-        c = c[Nx - 1 - maxlags:Nx + maxlags]
+        correls = correls[Nx - 1 - maxlags:Nx + maxlags]
 
         if usevlines:
-            a = self.vlines(lags, [0], c, **kwargs)
+            a = self.vlines(lags, [0], correls, **kwargs)
+            # Make label empty so only vertical lines get a legend entry
+            kwargs['label'] = ''
```

Comment:
This is the actual fix

Reply:
might be better to `kwargs.pop('label', '')` so that if the user has not passed in a label the `axhline` does the default thing?